### PR TITLE
Api 2.0 build docs pipeline

### PR DESCRIPTION
**Pull request by zhenmichael**
Updated build-docs pipeline to generate API json artifacts and markdown for both 2.0 and 1.0 content.

Currently, the 1.0 api content is generated locally by running api-extractor on the lts branch content and uploaded to the fluidframework container storage as 'latest-1.0.tar.gz'. The pipeline has been updated to download this blob before running the markdown documenter. 

In docs/package.json, a new build:multi-version script was added which calls the relevant build-docs scripts such as rollup-api-json and render-api-documentation with an additional parameter boolean parameter to signal the scripts to process content for previous version apis as well. (note that, the build-docs pipeline is currently set to run the old build script until 2.0 release)

Note that the previous branch versions are maintained in two separate areas since there is no straighforward way to import pipeline parameters from another yaml file. The first is docs/data/versions.yaml and the second is the parameters in build-docs.yaml. Some checks will need to be added to ensure that the two lists match up.

### CODE REVIEW

**Review by Josmithr — COMMENTED** [review]
Overall looking really good! I left a couple of comments we'll want to address before merging, but nothing too substantial.

The biggest thing, I think, is updating the pipeline to share a single file with the website build for designating the versions we care about.

File: tools/pipelines/build-docs.yml
Comment (Josmithr) [inline tools/pipelines/build-docs.yml:214]: Can we add some documentation at the root of this file explaining the assumptions we're making here? Devs looking at this pipeline in the future will need to know about the artifact scheme we're depending on here.

File: docs/package.json
Comment (zhenmichael) [reply docs/package.json]: I'm currently running `npm run build:md-magic` under `build:multi-version`. Would it be repeated if I run it on both, or should I move the call to `build:api:multi-version`?

Comment (Josmithr) [reply docs/package.json]: Sorry, I meant you should manually run the script and push the changes to the branch. You added new scripts to the package.json, so it will generate changes to the `scripts` table in the docs README.

File: docs/package.json
Comment (Josmithr) [inline docs/package.json:25]: Does this need to run both `build:api` _and_ `build:api:multi-version`? Shouldn't it only need the multi-version variant?

File: docs/package.json
Comment (Josmithr) [inline docs/package.json:28]: Per our offline discussion, it would be nice if we had something like a `download:multi` script to support being able to generate local website builds using the new infra.

File: docs/rollup-api-json.js
Comment (Josmithr) [inline docs/rollup-api-json.js:23]: Can we add a doc comment to this explaining what it does and what the parameters mean?

File: tools/pipelines/build-docs.yml
Comment (Josmithr) [inline tools/pipelines/build-docs.yml:5]: Nit: I would express these docs in terms of the "1.0" branch, rather than the "lts" branch. LTS will change over time, whereas it's the 1.0 branch specifically with shortcomings.

But otherwise, these docs look great! Thank you for the thorough explanation! This will make future devs' lives much easier when working in here.

File: tools/pipelines/build-docs.yml
Comment (Josmithr) [inline tools/pipelines/build-docs.yml:210]: Nit: I always recommend adding a brief comment above each task giving a quick explanation as to what is being done and why. In this case, just a quick note that we need to install jq for the subsequent step.

Comment (Josmithr) [reply tools/pipelines/build-docs.yml:210]: This way if other steps are changed in the future, it's easier to tell if this is still needed or not.

**Review by Josmithr — COMMENTED** [review]
Left a couple more comments, but this is looking really close.

Also, feel free to leave the `download:multi` work for a subsequent PR if you would prefer - since nothing is consuming this infra yet, I don't see any harm on splitting up work if that makes your life easier / helps unblock @chentong7's PR 🙂

File: docs/api-markdown-documenter/index.js
Comment (Josmithr) [inline docs/api-markdown-documenter/index.js:8]: Nit
```suggestion
 * in data/versions.json. This script allows for an optional boolean parameter which determines whether 
```

File: docs/readVersionData.js
Comment (Josmithr) [inline docs/readVersionData.js:21]: Nit:
```suggestion

return(console.log(parsedData.params.previousVersions.join('\n')));
```

File: docs/rollup-api-json-helper.js
Comment (Josmithr) [reply docs/rollup-api-json-helper.js:22]: Thanks! The docs look great!

**Review by Josmithr — APPROVED** [review]

File: docs/api-markdown-documenter/index.js
Comment (Josmithr) [inline docs/api-markdown-documenter/index.js:40]: Tong and I were looking at this on her PR, and it occurred to me that we should change the error handling here a bit. As it is, if 1 of our render passes fails, we exit the process immediately (hiding any other potential errors we could have found in other render passes). I would recommend something like the following:
```suggestion
docVersions.forEach((version) => {
	apiDocRenders.push(
		renderApiDocumentation(version).then(
			() => {
				console.log(chalk.green(`${version} API docs written!`));
			},
			(error) => {
				throw new error(`${version} API docs could not be written due to an error:`, error);
			},
		),
	);
});

Promise.all(apiDocRenders).then(
	() => {
		console.log(chalk.green("All API docs written!"));
		process.exit(0);
	},
	(error) => {
		console.error(error);
		process.exit(1);
	},
);
```

**Pull request merged**
